### ISSUE TITLE
Cannot create new shapes category

### ISSUE THREAD

**Issue by damithc** [PowerPointLabs/PowerPointLabs#1898]
5.0.0.0 (Dev-release)
Win 10, Office 2016

Steps: open shapes lab, right-click, and choose to create a new category.

Problem: I get this error no matter what name I give

![image](https://user-images.githubusercontent.com/1673303/56653285-ed727a00-66bf-11e9-8898-e7d7f3f5a1ed.png)

**Comment by damithc** [comment]
I'm also getting this error from time to time. When it happens, I'm left with an extra slide at the end of the slide deck.
![image](https://user-images.githubusercontent.com/1673303/56654522-dbdea180-66c2-11e9-9f08-2aca5b82d694.png)

**Comment by YuPeiHenry** [comment]
@damithc I can't seem to reproduce the error for being unable to create a new category (dev-release 5.0.0). The condition check for the error message would seem to be `if (string.IsNullOrWhiteSpace(name) || invalidChars.IsMatch(name))`. It would seem that a possible point of failure is the regex.

**Comment by damithc** [comment]
While trying to reproduce, I encountered a different problem in my other computer. There is no place to right-click!
![image](https://user-images.githubusercontent.com/1673303/57186642-94071800-6f15-11e9-85f8-09490056311c.png)

**Comment by damithc** [comment]
But I couldn't reproduce it in my home computer. Will try again in office computer.

**Comment by YuPeiHenry** [comment]
> While trying to reproduce, I encountered a different problem in my other computer. There is no place to right-click!

Think this is a sizing issue with the panel. Either the user has to increase the width of the panel a little, or the default width should be changed.

**Comment by damithc** [comment]
> But I couldn't reproduce it in my home computer.

Couldn't reproduce it in the office computer too. 🤔 


> Think this is a sizing issue with the panel. Either the user has to increase the width of the panel a little, or the default width should be changed.

That may not be apparent to the user. On a related noted, we can add more buttons in this area for 
1. Downloading more shapes
2. Manage shapes (i.e., same options as those in the the context menu)
3. Settings

**Comment by yuhongherald** [comment]
Sent by a user. Error details:

> The user wants to add a category but giving a name and clicking on the ok button does nothing.
> The user only can click cancel.

I am unable to replicate the bug, will be doing a bit more investigation.

**Comment by damithc** [comment]
On the right-click menu, `remove category` and `rename category` don't seem to work either. Nothing happens when I click on them.

**Comment by damithc** [comment]
@yuhongherald can take a look? While some behaviors mentioned cannot be reproduced, the following can be fixed:


> That may not be apparent to the user. On a related noted, we can add more buttons in this area for
> 
> 1. Downloading more shapes
> 2. Manage shapes (i.e., same options as those in the the context menu)
> 3. Settings
> 
> ![image](https://user-images.githubusercontent.com/1673303/57192733-364eec00-6f66-11e9-908f-2b9936c7a18c.png)



> On the right-click menu, `remove category` and `rename category` don't seem to work either. Nothing happens when I click on them.

**Comment by yuhongherald** [comment]
@damithc I don't seem to be able to replicate the bug. Can you provide more context?

**Comment by damithc** [comment]
yes, the bug is no longer reproducible. But we can improve the UI as suggested above.
